src/textual/widgets/_markdown.py

```diff
@@ -500,19 +501,35 @@ class MarkdownFence(MarkdownBlock):
     def __init__(self, markdown: Markdown, code: str, lexer: str) -> None:
         self.code = code
         self.lexer = lexer
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
```

Comment:
I feel that, ideally, there would be some method of allowing the developer using `Markdown` to specify their preference for the themes to use for light and dark mode; which might mean that there needs to be a pair of properties on the `Markdown` widget itself that supplies these.

Reply:
I think it's a nice idea to specify both light/dark themes. But would you do them as properties or rather simply pass them in `__init__`? I think that I would either:
* Have one `theme` property that the developer can set on the widget that does listen to light/dark changes.
* Specify on `__init__` a light/dark theme (with defaults) and listen to light/dark changes.

Reply:
I think, if I were approaching this myself, I'd have them be reactive properties on `Markdown` (as the public API for setting them). My thinking here being:

1. If someone were to change the light or dark themes for code on a "live" `Markdown`, we can watch for those changes and, if necessary (we're in dark mode, and the dark mode theme gets changed), we can query through all of the instances of `MarkdownFence` within us and get them to refresh (a pretty niche situation, I'd say, but not very costly to handle).
2. `MarkdownFence` takes a reference to its parent `Markdown` anyway, so when it needs to reference the appropriate theme, it can just reference the correct property on the markdown object.
3. `MarkdownFence` can watch for the chance in light/dark mode and switch to the most appropriate theme when necessary.

Hopefully that makes sense?